File: 0070-climbing-stairs/climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        table = []
        if n == 1:
            return 1
        table.append(1) # 0
        table.append(1) # 1
        table.append(2) # 2
        for i in range(2, n):
            table.append(table[i-1]+table[i])
        return table[-1]

File: 0070-climbing-stairs/climbing-stairs.cpp
class Solution {
    std::unordered_map<int, int> store;
public:
    int climbStairs(int n) {
        if (store.contains(n)) return store[n];
        store[1] = 1;
        store[2] = 2;

        for (int i = 3; i <= n; ++i) {
            store[i] = store[i-1] + store[i-2];
        }

        return store[n];
    }
};
